src/NzbDrone.Core/Download/Clients/Tribler/TriblerDownloadSettings.cs

```diff
@@ -0,0 +1,63 @@
+using System.Text.RegularExpressions;
+using FluentValidation;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.ThingiProvider;
+using NzbDrone.Core.Validation;
+using NzbDrone.Common.Extensions;
+
+namespace NzbDrone.Core.Download.Clients.Tribler
+{
+    public class TriblerSettingsValidator : AbstractValidator<TriblerDownloadSettings>
+    {
+        public TriblerSettingsValidator()
+        {
+            RuleFor(c => c.RootUrl).ValidRootUrl();
+
+            RuleFor(c => c.ApiKey).NotEmpty();
+
+            RuleFor(c => c.TvCategory).Matches(@"^\.?[-a-z]*$", RegexOptions.IgnoreCase).WithMessage("Allowed characters a-z and -");
+
+            RuleFor(c => c.TvCategory).Empty()
+                .When(c => c.TvDirectory.IsNotNullOrWhiteSpace())
+                .WithMessage("Cannot use Category and Directory");
+
+            RuleFor(c => c.AnonymityLevel).GreaterThanOrEqualTo(0).WithMessage("Should be equal or more than zero.");
+
+        }
+    }
+
+    public class TriblerDownloadSettings : IProviderConfig
+    {
+        private static readonly TriblerSettingsValidator Validator = new TriblerSettingsValidator();
+
+        public TriblerDownloadSettings()
+        {
+            RootUrl = "http://localhost:52194";
```

Comment:
For better or worse download clients separate protocol, host and port into separate fields.